src/components/map/MapService.js

```diff
@@ -371,12 +371,51 @@ goog.require('ga_urlutils_service');
    */
   module.provider('gaLayers', function() {
 
-    this.$get = function($http, $q, $rootScope, $translate, $window,
+    this.$get = function($http, $q, $rootScope, $translate, $window, $timeout,
         gaBrowserSniffer, gaDefinePropertiesForLayer, gaMapUtils,
         gaNetworkStatus, gaStorage, gaTileGrid, gaUrlUtils,
         gaStylesFromLiterals, gaGlobalOptions, gaPermalink, 
         gaLang, gaTime) {
 
+      var MAX_RETRIES = 100;
+      var RETRIES_INTERVAL = 5000;
+
+      var retryOnError = function(msg, el, olsource) {
+        if (angular.isDefined(msg) &&
+            angular.isDefined(olsource)) {
+          olsource.on(msg, function(event) {
+            if (event[el]._retries == undefined) {
+              event[el]._retries = 0;
+            }
+            event[el]._retries++;
+            if (event[el]._retries <= MAX_RETRIES) {
+              $timeout(function() {
+                event[el].state = 0;
+                event[el].load();
+              }, RETRIES_INTERVAL);
```

Comment:
Set the 3rd parameter to false to avoid triggering a digest cycle on each failure: `, RETRIES_INTERVAL, false) `